### PR TITLE
rewriting_trending_from_mip6_into_mip5 for delivery reports

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
…azer

### CODE REVIEW

File: trailblazer/mip/trending.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        qcmetrics_raw (dict): raw YAML input from MIP analysis qc metric file
+        sampleinfo_raw (dict): raw YAML input from MIP analysis qc sample info file
+    Returns:
+        dict: parsed data
+    """
+
+    ### Define output dict
+    outdata = {
+        'analysis_sex': {},
+        'at_dropout': {},
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: We could comment this one out

File: trailblazer/mip/trending.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        'family': None,
+        'duplicates': {},
+        'gc_dropout': {},
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: This one as well

File: trailblazer/mip/trending.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        'genome_build': None,
+        'insert_size_standard_deviation': {},
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: And this one

File: trailblazer/mip/trending.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        'mapped_reads': {},
+        'median_insert_size': {},
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: And this one

File: trailblazer/mip/trending.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    Args:
+        mip_config_raw (dict): raw YAML input from MIP analysis config file
+        qcmetrics_raw (dict): raw YAML input from MIP analysis qc metric file
+        sampleinfo_raw (dict): raw YAML input from MIP analysis qc sample info file
+    Returns:
+        dict: parsed data
+    """
+
+    ### Define output dict
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: What's up with these triple `#`?

File: trailblazer/mip/trending.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        'gc_dropout': {},
+        'genome_build': None,
+        'insert_size_standard_deviation': {},
+        'mapped_reads': {},
+        'median_insert_size': {},
+        'mip_version': None,
+        'sample_ids': [],
+    }
+    ### Config
+    ## Parse raw mip_config into dict
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: What's up with these dual `#`?

File: trailblazer/mip/trending.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        outdata['analysis_sex'][sample_data['id']] = sample_data['predicted_sex']
+
+    ### Qc sample info
+    ## Parse qc sample info file
+    sampleinfo_data = files.parse_sampleinfo(sampleinfo_raw)
+
+    ## Add mip version
+    outdata['mip_version'] = sampleinfo_data['version']
+
+    ## Add mip version
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: `# Add genome build`

File: trailblazer/mip/trending.py (hunk 1 above)
Comment (henrikstranneheim) [reply trailblazer/mip/trending.py]: This is an important qc-metric to trend as was decided previously. Are you adding this info from some other source?
I think it is bad practise to leave lines of code that are not used in a "commented form". It clutteres the code and makes it more complicated than it is. Removing the unused line is prefered and then adding it in when required. In this case I do not really understand why you want to comment it out as you can just ignore this key when parsing the output in the downstream process (if it is not ready for the "at_dropout"), but I am not up to speed on what you are planning here.

File: trailblazer/mip/trending.py (hunk 2 above)
Comment (henrikstranneheim) [reply trailblazer/mip/trending.py]: This is an important qc-metric to trend as was decided previously. Are you adding this info from some other source?

File: trailblazer/mip/trending.py (hunk 3 above)
Comment (henrikstranneheim) [reply trailblazer/mip/trending.py]: This is an important qc-metric to trend as was decided previously. Are you adding this info from some other source?

File: trailblazer/mip/trending.py (hunk 4 above)
Comment (henrikstranneheim) [reply trailblazer/mip/trending.py]: This is an important qc-metric to trend as was decided previously. Are you adding this info from some other source?

File: trailblazer/mip/trending.py (hunk 5 above)
Comment (henrikstranneheim) [reply trailblazer/mip/trending.py]: It's just a way of commenting what the next block of code is handling. If you have another system in place in trailblazer feel free to modify accordingly.

File: trailblazer/mip/trending.py (hunk 6 above)
Comment (henrikstranneheim) [reply trailblazer/mip/trending.py]: see above

File: trailblazer/mip/trending.py (hunk 1 above)
Comment (henrikstranneheim) [reply trailblazer/mip/trending.py]: Ok, now I see you are just trying to get it to be compatible with the mip5.0 in production, right?

Comment (ingkebil) [reply trailblazer/mip/trending.py]: ... like the title said ;)

**Comment by patrikgrenfeldt** [comment]
All requested changes performed, what do you think of it now?

**Comment by henrikstranneheim** [comment]
Wow, You really went all in for utility subs, 😁. Very nice and clear! 👍

**Comment by ingkebil** [comment]
Let's merge and :ship:

File: trailblazer/mip/trending.py (hunk 5 above)
Comment (ingkebil) [reply trailblazer/mip/trending.py]: Python has a style guide which we are following: https://www.python.org/dev/peps/pep-0008/
e.g. Pycharm automatically highlights any deviations from pep-8.

**Pull request merged**